### DISCUSSION TITLE
Does the number of collections per db has a limit？

### DISCUSSION THREAD

**Discussion by StormstoutY** [milvus-io/milvus#27523] [Q&A and General discussion] [answered]
From the docs https://milvus.io/docs/manage_databases.md，I notice that a Milvus cluster supports a maximum of 64 databases. Then I wonder how much collections does a db supporting to create? If there is indeed a limit on the number of collections created, how can I adjust the upper limit?

**Comment by yhmo** [comment, marked as answer]
Max collections number and max collections number for each db are defined in the milvus.yaml:
https://github.com/milvus-io/milvus/blob/a71516530692cff2547efb9a28824d5bcf0c24a9/configs/milvus.yaml#L542C33-L542C33

The default limit is 65536.
In fact, if you created thousands of collections in milvus, you will see the milvus server become slower and slower, because the server need to observe/monitor each collection even the collection is empty.

**Reply by StormstoutY** [reply]
Thank you for your reply. In fact, I want to solve the problem of using milvus with multi-tenants. I saw this official solution（https://github.com/milvus-io/milvus/issues/23553  https://z2-cn-dev.zilliz.cc/blog/milvus-partitionkey-feature）, but it may still be different from our scenario. Will milvus make more optimizations for this scenario in the future?

**Reply by yhmo** [reply]
Yes, multi-tenants is an important ability for some users, we are continually optimizing it, reducing the required resource of each collection so that it can support more partitions/collections.

**Comment by gustavz** [comment]
@yhmo is there any update wrt collection limits? Are there any workaround for multi tenant applications that want to serve many thousand and continuously growing number of collections?

**Reply by yhmo** [reply]
We have made some optimizations in the recent minor versions. In our test, 10000 collections, milvus server is more stable than before, but the server is still response slowly.
